channels/generic/http.py

```diff
@@ -1,31 +1,17 @@
 from channels.consumer import AsyncConsumer
 
+from ..exceptions import (
+    StopConsumer,
+)
+
 
 class AsyncHttpConsumer(AsyncConsumer):
     """
     Async HTTP consumer. Provides basic primitives for building asynchronous
     HTTP endpoints.
     """
 
-    async def __call__(self, receive, send):
-        """
-        Async entrypoint - concatenates body fragments and hands off control
-        to ``self.handle`` when the body has been completely received.
-        """
-        self.send = send
-        body = []
-        try:
-            while True:
-                message = await receive()
-                if message["type"] == "http.disconnect":
-                    return
-                else:
-                    if "body" in message:
-                        body.append(message["body"])
-                    if not message.get("more_body"):
-                        await self.handle(b"".join(body))
-        finally:
-            await self.disconnect()
+    body = []
```

Comment:
This variable will only be initialized once per Python process instead of once per request. 

Suggestion: You could move the initialization inside `async def __call__` (`self.body = []`) and call `super().__call__` so that `AsyncConsumer` can do its thing.